libs/agno/agno/cli/auth_server.py

```diff
@@ -20,11 +27,74 @@ def _set_response(self):
         self.send_header("Access-Control-Allow-Headers", "*")
         self.send_header("Access-Control-Allow-Methods", "POST")
         self.end_headers()
+    
+    def _set_html_response(self, html_content: str, status_code: int = 200):
+        """Set the response headers and content type to HTML."""
+        self.send_response(status_code)
+        self.send_header("Content-type", "text/html")
+        self.send_header("Access-Control-Allow-Origin", "*")
+        self.send_header("Access-Control-Allow-Headers", "*")
+        self.send_header("Access-Control-Allow-Methods", "POST, GET, OPTIONS")
+        self.send_header("Access-Control-Allow-Credentials", "true")
+        self.end_headers()
+        self.wfile.write(html_content.encode("utf-8"))
+
+    
+    
+    def do_GET(self):
+        """Handle the redirect from the browser with auth cookies."""
+        
+        # If path contains success or error parameters, show appropriate page
+        if "?result=success" in self.path:
+            self._set_html_response(get_success_page(), status_code=200)
+            # Shutdown server after showing success
+            self.server.running = False
+            return
+        elif "?result=error" in self.path:
+            error_type = self.path.split("error_type=")[1].split("&")[0] if "error_type=" in self.path else ""
+            if error_type == "no_cookies":
+                self._set_html_response(get_no_cookies_error_page(), status_code=400)
+            else:
+                self._set_html_response(get_no_auth_token_error_page(), status_code=400)
+            return
+        
+        # Extract cookies from the request when processing
+        cookies_header = self.headers.get('Cookie')
+        
+        # If no cookies, redirect to error page
+        if not cookies_header:
+            # First show loading page
+            self._show_loading_with_redirect("?result=error&error_type=no_cookies")
+            return
+            
+        # Parse cookies - find the session or auth cookie
+        cookies = {}
+        for cookie in cookies_header.split(';'):
+            if '=' in cookie:
+                name, value = cookie.strip().split('=', 1)
+                cookies[name] = value
+        
+        # Find your specific session cookie
+        auth_token = cookies.get('__agno_session')
```

Comment:
Let's get the cookie name from the settings. It will probably change soon
```python
from api.settings import api_settings

auth_token = cookies.get(api_settings.user_auth_token_cookie)
```